### DISCUSSION TITLE
Update archive directory after filename format change or tag name change?

### DISCUSSION THREAD

**Discussion by endotronic** [jonaswinkler/paperless-ng#749] [Support] [answered]
I really appreciate that there is an option to configure the directory structure and filenames for the document archive. This is very important to me, since one day I might not use paperless-ng, and I would hate to have all the organization I've done stuck in the paperless-ng database.

I use the tag list as a directory in the configured structure. I noticed that if I change the name of a tag, it does not update what is already stored to disk; rather, it only applies to new documents or updates a document when it is changed. This is also true when I change the configuration for the directory and filename structure.

Is it possible to resync the whole document archive, basically renaming files and recreating directories as needed? The only way I have found to do this is to change something on every document, e.g. apply and remove a new tag.

Side note, and off topic: I would really like it if paperless-ng would set the dates on the files in the filesystem. They do not match the dates I have set in the document metadata in paperless-ng.

**Comment by jonaswinkler** [comment, marked as answer]
Hello!

The management command `document_renamer` goes through all documents and adjusts the filenames on disk if necessary. Also see https://paperless-ng.readthedocs.io/en/latest/administration.html#managing-filenames.

* This is generally required after changing `PAPERLESS_FILENAME_FORMAT`.
* Changing names of tags, correspondents and types should update filenames, that's an oversight on my part.

**Reply by endotronic** [reply]
Thanks so much for the quick reply! `document_renamer` is perfect for me, thanks!

**Reply by m0veax** [reply]
Hey, sorry for writing this in a solved thread, but somehow i can't find the command. Where do i have to start it? On cli? Or in the Administration UI?

**Reply by stueja** [reply]
@m0veax , please see 
https://paperless-ng.readthedocs.io/en/latest/administration.html#management-utilities
```
cd /path/to/paperless
docker-compose exec webserver <command> <arguments>
```

**Comment by stueja** [comment]
I have 1400 documents in a plain directory without subfolders. If I now change the `PAPERLESS_FILENAME_FORMAT` to the example `PAPERLESS_FILENAME_FORMAT={created_year}/{correspondent}/{title}` and run `document_renamer`, what will happen to those documents which do not have a `{correspondent}` set in the UI?

**Reply by Spoker** [reply]
correspondent will be "none" and files will be placed in {created_year}/none/{title}

**Reply by stueja** [reply]
Excellent, thank you!
